R/helper_functions.R

```diff
@@ -87,39 +89,45 @@ check_caretList_classes <- function(list_of_models) {
 #' @param list_of_models a list of caret models to check
 #' @importFrom caret modelLookup
 check_caretList_model_types <- function(list_of_models) {
-  #Check that models have the same type
+  # Check that models have the same type
   types <- sapply(list_of_models, function(x) x$modelType)
   type <- types[1]
-  stopifnot(all(types==type)) #TODO: Maybe in the future we can combine reg and class models
+  stopifnot(all(types == type)) # TODO: Maybe in the future we can combine reg and class models
 
-  #Check that the model type is VALID
+  # Check that the model type is VALID
   stopifnot(all(types %in% c("Classification", "Regression")))
 
-  #Warn that we haven"t yet implemented multiclass models
-  # add a check that if this is null you didn"t set savePredictions in the trainControl
-  #TODO: add support for non-prob models (e.g. rFerns)
-  if (type=="Classification" & length(unique(list_of_models[[1]]$pred$obs))!=2) {
-    if(is.null(unique(list_of_models[[1]]$pred$obs))) {
-      stop("No predictions saved by train. Please re-run models with trainControl set with savePredictions = TRUE.")
-    } else {
-      stop("Not yet implemented for multiclass problems")
+  # Warn that we have not yet implemented multiclass models
+  # add a check that if this is null you did not set savePredictions in the trainControl
+  # TODO: add support for non-prob models (e.g. rFerns)
```

Comment:
Small refactor here: this is more than just linting. Logic should work the same though